sirplatform.h

```diff
@@ -32,6 +32,17 @@
 #  else
 #   define __HAVE_ATOMIC_H__
 #  endif
+#  if defined(__GNUC__)
+#   if __GNUC__ <= 4
+#    if defined(__GNUC_MINOR__)
+#     if __GNUC_MINOR__ <= 8
+#      if !defined(__clang_version__)
+#       undef __HAVE_ATOMIC_H__
```

Comment:
No ... I had to do that because GCC 4.8 actually indicates via the macros it has the atomic functions but it really doesn't.  That was fixed in 4.9.  (I used Ubuntu 14.04 to test which has GCC 4.8.4).

Also, Clang masquerades as GCC by defining various things (like __GNUC__), and the old version of Clang that has masqueraded as GCC 4.8 worked fine with the atomics.

Essentially, I'm doing this test to avoid using stdatomic on actual versions of GCC below 4.9, regardless of how they incorrectly set the other feature detection macros.